runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -274,6 +276,8 @@ private ChannelInitializer getChannelInitializer() {
             protected void initChannel(@Nonnull Channel ch) throws Exception {
                 ch.pipeline().addLast(new IdleStateHandler(parameters.getIdleConnectionTimeout(),
                         parameters.getKeepAlivePeriod(), 0));
+                ch.pipeline().addLast(new Lz4FrameDecoder());
+                ch.pipeline().addLast(new Lz4FrameEncoder());
```

Comment:
This doesn't make sense: compression after encryption will probably not benefit from size reduction (since tls will produce a "random" stream of bytes). Change the order Compress -> encrypt.